File: platforms/macos/src/Core/PopupSizing.swift
import Foundation

/// Pure decision for the translation popup's fixed size (spec §8, conformance `popup-sizing`).
///
/// Two sizes: `normal`, and `large` = `largeWidthFactor` × normal width by `largeHeightFactor` ×
/// normal height. The popup uses `large` when the currently displayed entry's **source** text
/// length exceeds `largeSourceCharThreshold`. Keeping this as a pure, platform-neutral function lets
/// the rule be pinned by a shared conformance vector while each platform owns the actual pixels.
public enum PopupSizing {
    /// Source length strictly greater than this → large size. (`> 500`, so 500 is still normal.)
    public static let largeSourceCharThreshold = 500
    /// Large width = this × normal width.
    public static let largeWidthFactor = 2.0
    /// Large height = this × normal height.
    public static let largeHeightFactor = 1.5

    /// "normal" or "large" for the given source character count.
    public static func sizeClass(sourceChars: Int) -> String {
        sourceChars > largeSourceCharThreshold ? "large" : "normal"
    }
}

File: platforms/windows/src/TranslateTheDamn.Core/PopupSizing.cs
namespace TranslateTheDamn.Core;

/// <summary>
/// Pure decision for the translation popup's fixed size (spec §8, conformance <c>popup-sizing</c>).
///
/// Two sizes: <c>normal</c>, and <c>large</c> = <see cref="LargeWidthFactor"/> × normal width by
/// <see cref="LargeHeightFactor"/> × normal height. The popup uses <c>large</c> when the currently
/// displayed entry's <b>source</b> text length exceeds <see cref="LargeSourceCharThreshold"/>. Keeping
/// this as a pure, platform-neutral function lets the rule be pinned by a shared conformance vector
/// while each platform owns the actual pixels. Mirrors macOS <c>PopupSizing.swift</c>.
/// </summary>
public static class PopupSizing
{
    /// <summary>Source length strictly greater than this → large size. (<c>&gt; 500</c>, so 500 is still normal.)</summary>
    public const int LargeSourceCharThreshold = 500;

    /// <summary>Large width = this × normal width.</summary>
    public const double LargeWidthFactor = 2.0;

    /// <summary>Large height = this × normal height.</summary>
    public const double LargeHeightFactor = 1.5;

    /// <summary>"normal" or "large" for the given source character count.</summary>
    public static string SizeClass(int sourceChars) =>
        sourceChars > LargeSourceCharThreshold ? "large" : "normal";
}
